lib/grandpa/message_tracker.go

```diff
@@ -108,18 +102,17 @@ func (t *tracker) handleBlock(b *types.Block) {
 	defer t.mapLock.Unlock()
 
 	h := b.Header.Hash()
-	if vms, has := t.voteMessages[h]; has {
-		for _, v := range vms {
-			// handleMessage would never error for vote message
-			_, err := t.handler.handleMessage(v.from, v.msg)
-			if err != nil {
-				logger.Warnf("failed to handle vote message %v: %s", v, err)
-			}
+	vms := t.votes.getMessagesForBlockHash(h)
+	for _, v := range vms {
+		// handleMessage would never error for vote message
+		_, err := t.handler.handleMessage(v.from, v.msg)
+		if err != nil {
+			logger.Warnf("failed to handle vote message %v: %s", v, err)
 		}
-
-		delete(t.voteMessages, h)
 	}
 
+	t.votes.delete(h)
+
```

Comment:
Comment here since I got self-confused reading the diffs;
- before we would delete only if the block hash existed
- now we delete the block hash from the votes tracker whether it exists or not

Reply:
Thanks for this!